### PR TITLE
Add support for providing an existing secret to internal database

### PR DESCRIPTION
**Pull request by cten**
This PR provides the means to use an external tool to create/manage secrets that then can be used by harbor internal database instead of relying on helm auto-generated password.

### CODE REVIEW

**Review by Vad1mo — APPROVED** [review]
LGTM

File: Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: harbor
-version: 1.4.0-dev
+version: 1.4.1-dev
```
Comment (zyyw) [inline Chart.yaml:3]: please do not change this.

File: templates/core/core-secret.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ data:
   {{- if not .Values.existingSecretAdminPassword }}
   HARBOR_ADMIN_PASSWORD: {{ .Values.harborAdminPassword | b64enc | quote }}
   {{- end }}
-  {{- if not .Values.database.external.existingSecret }}
+  {{- if and (not .Values.database.external.existingSecret) (not .Values.database.internal.existingSecret) }}
```
Comment (zyyw) [inline templates/core/core-secret.yaml:24]: need to update `core-dpl.yaml` as well

File: templates/exporter/exporter-dpl.yaml [hunk 3]
```diff
@@ -82,6 +82,13 @@ spec:
               name: {{ .Values.database.external.existingSecret }}
               key: password
         {{- end }}
+        {{- if .Values.database.internal.existingSecret }}
```
Comment (zyyw) [inline templates/exporter/exporter-dpl.yaml:85]: may need to change `{{- if .Values.database.internal.existingSecret }}` to `{{- if and (eq .Values.database.type "internal") .Values.database.internal.existingSecret -}}`

Comment (zyyw) [reply templates/exporter/exporter-dpl.yaml:85]: also may also update line 78, `{{- if .Values.database.external.existingSecret }}` -> `{{- if and (eq .Values.database.type "external") .Values.database.external.existingSecret }}`

**Comment by cten** [comment]
I had issues with merging upstream changes, so I recreated the PR here https://github.com/goharbor/harbor-helm/pull/1733

**Pull request closed without merging**